src/js/visual/configurator-csr.js

```diff
@@ -1703,6 +1717,7 @@ ripe.ConfiguratorCsr.prototype._onPreConfig = function(self) {
  * @ignore
  */
 ripe.ConfiguratorCsr.prototype._onPostConfig = async function(self, config) {
+    if (!this.owner.hasStrategy("csr")) return;

```

Comment:
Shouldn't this not be here? This just tells me there is a bug somewhere, otherwise this kind of guard should be anywhere and everywhere which is why this is smelling like a bug somewhere.

Reply:
This is a guard to fix the following problem:

When you have a configured model e.g a `dummy` `cube` and you call `ripe.config` to change to a different model e.g `dummy` `bill` there is a chain of events that's triggered. That chain of events includes the events `pre_config` `config` and `post_config`. Only after `post_config` is triggered you get to know if strategy `csr` is available for the configured build.

Since `_onPostConfig` was binded in a model that has CSR, when you change to a model that doesn't have CSR, `_onPostConfig` will get called and you have no way to prevent this from happening. 

Here's the debug log showing that CSR's `_onPostConfig` will happen before `_unregisterConfigGlobalHandlers` and we have no way of preventing this because the unmount of the CSR configurator takes more time than the `_onPostConfig`.

![image](https://user-images.githubusercontent.com/24736423/223425386-468f6474-0949-4d8e-8498-2c1fbd77cfa5.png)

Users should be able to smoothly switch between different models but with the current CSR implementation this error will be thrown. I think `_onPostConfig` should not be running for strategies that don't include "CSR". In the current implementation `_onPostConfig` is running for any strategy after `ripe.config` so I think this guard should be in RIPE SDK and this problem should be dealt here.

Reply:
I think it's working as intended then. If the model doesn't have CSR and you are trying to use CSR it should throw errors.
The one using this configurate is the one who should handle this, as in, it should deinit the configurator as it's not going to use it.
It's the same behaviour has having 3 thumbnails in a model and changing to a model that has only 2 thumbnails, you have to `deinit()` the extra one or it will error it out too.

Think like this, if you invert the roles and the norm was to have every build configurated to CSR instead of PRC and PRC were the ones that are only occasional configurated, then this change would be done there and not here which would not be correct to. A configurator is just another type of Visual component. It has the same weight as any other kind, as in, `configurator-prc`, `configurator-prc` and `image` have equal footing in the SDK.

Reply:
> I think it's working as intended then.

Considering that we support switching models without having to do `configurator.deinit` how do you suggest for an implementation to do deal with this without throwing errors? If you wish you can share an example.html like before.

> Think like this, if you invert the roles and the norm was to have every build configurated to CSR instead of PRC and PRC were the ones that are only occasional configurated, then this change would be done there and not here

I think this change can be done in both places if we don't want to assume that there is always a PRC mode which I think is currently the case.
In the future we may have CSR only models and we might find more problems related to PRC not doing some deinits as it should.

Reply:
We support changing to builds that have those capabilites. If you change to something unsupported it will throw erros. For example thumbnails.

As you can see, in White, the thumbnails destroy themselves when their number change for example:
https://github.com/ripe-tech/ripe-commons-pluginus/blob/f5fbdf5205ede56cfe5e5e7e8f19c00eb6cbab5a/vue/components/atoms/thumbnail/thumbnail.vue#L129

Reply:
> the thumbnails destroy themselves when their number change

It's much lighter to destroy an image bind than to destroy a configurator bind so I don't think this is a fair comparison.

I think we should still support changing to builds that have different capabilities but given that white continues to work without this I'll resolve.

If a critical request comes up I'll bring this topic again.